app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt

```diff
@@ -0,0 +1,480 @@
+            val decimal = value / DECIMAL_CONVERSION
+            val textSize = TEXT_SIZE * decimal
+            mExampleTextTextView?.setTextSize(TypedValue.COMPLEX_UNIT_SP, textSize)
+        }
+    }
+
+    override fun onSaveInstanceState(): Parcelable {
+        val superState = super.onSaveInstanceState()
+        if (isPersistent) {
+            // No need to save instance state since it's persistent
+            return superState
+        }
+
+        // Save the instance state
+        val myState = SavedState(superState)
+        myState.mSeekBarValue = mSeekBarValue
+        myState.mMin = mMin
+        myState.mMax = mMax
+        return myState
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        if (state?.javaClass != SavedState::class.java) {
+            // Didn't save state for us in onSaveInstanceState
+            super.onRestoreInstanceState(state)
+            return
+        }
+
+        // Restore the instance state
+        val myState = state as SavedState?
+        super.onRestoreInstanceState(myState!!.superState)
+        mSeekBarValue = myState.mSeekBarValue
+        mMin = myState.mMin
+        mMax = myState.mMax
+        notifyChanged()
+    }
+
+    /**
+     * SavedState, a subclass of [BaseSavedState], will store the state of this preference.
+     *
+     *
+     * It is important to always call through to super methods.
+     */
+    private class SavedState : BaseSavedState {
+
+        internal var mSeekBarValue: Int = 0
+        internal var mMin: Int = 0
+        internal var mMax: Int = 0
+
+        internal constructor(source: Parcel) : super(source) {
+
+            // Restore the click counter
+            mSeekBarValue = source.readInt()
+            mMin = source.readInt()
+            mMax = source.readInt()
+        }
+
+        internal constructor(superState: Parcelable) : super(superState)
+
+        override fun writeToParcel(dest: Parcel, flags: Int) {
+            super.writeToParcel(dest, flags)
+
+            // Save the click counter
+            dest.writeInt(mSeekBarValue)
+            dest.writeInt(mMin)
+            dest.writeInt(mMax)
+        }
+
+        companion object {
+            @JvmField
+            val CREATOR: Parcelable.Creator<SavedState> = object : Parcelable.Creator<SavedState> {
+                override fun createFromParcel(`in`: Parcel): SavedState {
+                    return SavedState(`in`)
+                }
+
+                override fun newArray(size: Int): Array<SavedState?> {
+                    return arrayOfNulls(size)
+                }
+            }
+        }
+    }
+
+    companion object {
+        private const val TAG = "SeekBarPreference"
+        private const val STEP_SIZE = 5
+        private const val MIN_VALUE = 50
+        private const val DECIMAL_CONVERSION = 100f
+        private const val TEXT_SIZE = 16f
+        private const val SEEK_BAR_MAX = 100
+    }
```

Comment:
In java version these fields are private, therefore there is no need to define it inside `companion object`.